crates/lib/src/pantry_db.rs

```diff
@@ -103,6 +115,24 @@ pub fn which(cmd: &String, conn: &Connection) -> Result<Vec<String>, rusqlite::E
     Ok(rv)
 }
 
+pub fn projects_for_symbol(
+    symbol: &String,
+    conn: &Connection,
+) -> Result<Vec<String>, rusqlite::Error> {
+    let mut stmt = conn.prepare(
+        "
+        SELECT project FROM provides WHERE program = ?1
+        UNION
+        SELECT project FROM aliases WHERE alias = ?1;",
+    )?;
+    let mut rv = Vec::new();
+    let mut rows = stmt.query(params![symbol])?;
+    while let Some(row) = rows.next()? {
+        rv.push(row.get(0)?);
+    }
```

Comment:
just for style reasons, this is:
```rust
Ok(
    stmt
    .query(params![symbol])?
    .map(|row| row.get(0).expect("Query returned malformed data."))
    .collect::<Vec<String>
)
```
assuming the borrow checker doesn't think you're dropping bits.

Reply:
Doesn't compile and I am not enough of a rust guru yet to fix

Reply:
if you haven't, give `pkgx bacon -j clippy` a try ;)

Reply:
I have not, though I have been there numerous times lately since it is the latest addition to the pantry. I guess I don't see how it would help me here.

Reply:
not here specifically, but the feedback is rapid and, to my mind, better organized than rust-analyzer+vscode (it also works no matter what base directory you're in, which rust-analyzer struggles with).